test/test_fat.rb

```diff
@@ -269,7 +269,7 @@ def test_change_dylib_id
       really_big_id = "x" * 4096
 
       # test failsafe for excessively large IDs (w/ no special linking)
-      assert_raises MachO::HeaderPadError do
+      assert_raises MachO::MachOSliceModificationError do
```

Comment:
This needs to be reverted as this test should always result in a fatal `HeaderPadError` exception.